### DISCUSSION TITLE
Can we retrieve the execution time of assets using Op Hooks?

### DISCUSSION THREAD

**Discussion by s000ne** [dagster-io/dagster#21771] [Q&A]
I'm considering using Op Hooks to notify Slack with a message at the end of each asset execution. Is there a way to retrieve the execution time of assets using Op Hooks?

**Comment by garethbrickman** [comment]
Execution time isn't among the available [HookContext properties](https://docs.dagster.io/concepts/ops-jobs-graphs/op-hooks#hook-context). You would either have to use the information available to find/calculate the execution time within the Op Hook's code, otherwise you can provide information from the asset to the Op Hook via op_tags. There's [a nice example of that here.](https://github.com/dagster-io/dagster/issues/8577#issuecomment-1730507416)
